File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int n=nums.length;
        int i=0,j=n-1;
        while(i<=j)
        {
            int mid=(i+j)/2;
            if(nums[mid]>target)
            {
                j=mid-1;
            }
            else if(nums[mid]<target)
            {
                i=mid+1;
            }   
            else
            {
                return mid;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.cpp
#include <vector>
#include <numeric>
using namespace std;


class Solution {
public:
    bool check(int target,int sum,int i,vector<int>& nums,int n)
    {
        if(i>=n)
        {
            return false;
        }
        if(target==sum)
        {
            return true;
        }
        return check(target+nums[i],sum-nums[i],i+1,nums,n) || check(target,sum,i+1,nums,n);    
    }
    bool canPartition(vector<int>& nums) {
        int sum=accumulate(begin(nums),end(nums),0);
        return check(0,sum,0,nums,nums.size());
    }
};
